src/test/regress/sql/multi_alter_table_add_constraints_without_name.sql

```diff
@@ -0,0 +1,121 @@
+--
```

Comment:
A new test file sounds good, and can be extended to future work on general no-name constraints.

I think you can connect to the worker in every subtest here, including the name collision subtest and the reference table